src/components/Post/Post.jsx

```diff
@@ -4,52 +4,337 @@ import {
+                fullWidth
+              />
+            </Box>
+
+            <Button onClick={handleReply}>Reply</Button>
+          </Stack>
+
+          <Stack>
+            {post?.data?.comments.map((comment) => {
+              const commentByUser = users.find(
+                (user) => user.id === comment.commentBy
+              );
+              return (
+                <ListItem direction="row" key={comment.commentId}>
+                  <Button>
+                    <Avatar src={commentByUser.data.profilePicture} />
+                  </Button>
+
+                  <Box>
+                    <Typography variant="caption">
+                      {commentByUser.data.email}
+                    </Typography>
+                    <Typography variant="body1">
+                      {comment.commentText}
+                    </Typography>
+                  </Box>
+                </ListItem>
+              );
+            })}
+          </Stack>
+        </Box>
+      )}
+      <Divider />
+    </Box>
+  );
+};
+
+const MenuPanel = ({ post, handleDeletePost }) => {
+  const [anchorEl, setAnchorEl] = useState(null);
+  const open = Boolean(anchorEl);
+  const handleClick = (event) => {
+    setAnchorEl(event.currentTarget);
+  };
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+  const handleDelete = () => {
+    handleClose();
+    handleDeletePost();
+  };
+  return (
+    <div>
+      <IconButton
+        size="small"
+        sx={{ alignItems: "flex-start" }}
+        aria-controls={open ? "basic-menu" : undefined}
+        aria-haspopup="true"
+        aria-expanded={open ? "true" : undefined}
+        onClick={handleClick}
+      >
+        <MoreVertIcon />
+      </IconButton>
+      <Menu
+        id="basic-menu"
+        anchorEl={anchorEl}
+        open={open}
+        onClose={handleClose}
+        MenuListProps={{
+          "aria-labelledby": "basic-button",
+        }}
+      >
+        <MenuItem>
+          <EditPostModal post={post} handleMenuClose={handleClose} />
+        </MenuItem>
+        <MenuItem onClick={handleDelete}>Delete</MenuItem>
+      </Menu>
+    </div>
+  );
+};
+
+const style = {
+  position: "absolute",
+  top: "50%",
+  left: "50%",
+  transform: "translate(-50%, -50%)",
+  width: 400,
+  bgcolor: "background.paper",
+  border: "2px solid #000",
+  boxShadow: 24,
+  p: 4,
+};
+
+const EditPostModal = ({ handleMenuClose, post }) => {
+  const [open, setOpen] = useState(false);
+  const handleOpen = () => setOpen(true);
+  const handleClose = () => {
+    handleMenuClose();
+    setOpen(false);
+  };
+
+  return (
+    <div>
+      <div onClick={handleOpen}>Edit</div>
+      <Modal
+        open={open}
+        onClose={handleClose}
+        aria-labelledby="modal-modal-title"
+        aria-describedby="modal-modal-description"
+      >
+        <Box sx={style}>
+          <EditPost post={post} />
+        </Box>
+      </Modal>
+    </div>
   );
 };
```

Comment:
Better to move these components to seperate files

Reply:
Thanks for the suggestion, will do it later